src/nvim/ex_cmds2.h

```diff
@@ -16,6 +16,32 @@
 #define CCGD_ALLBUF     8       // may write all buffers
 #define CCGD_EXCMD      16      // may suggest using !
 
+/// Growarray to store info about already sourced scripts.
+/// Also store the dev/ino, so that we don't have to stat() each
+/// script when going through the list.
+typedef struct scriptitem_S {
```

Comment:
does this still need to be moved?

Reply:
I'll let you know once I can remember why I did it in the first place.

Reply:
We can keep the `struct scriptitem_S ` isolated in the type definition for `scriptitem_T` needs to be in `ex_cmd2.h` as the `get_cureent_script_id` returns a `scriptitem_T*` without the typedef in `ex_cmd2.h` the function declaration in `ex_cmd2.c.generated.h` won't compile .

So I've left the typedef in the `ex_cmd.h` file and moved the struct defination in `ex_cmd2.c` . With that functions outside of `ex_cmd2.c` will be able to pass around a `scriptitem_T` but only function inside `ex_cmd2.c` will be able to inspect and edit it's contents .

We can consider this resolved ?

I'm no expert in c let me know if there's a  better way to solve this.